src/ray/raylet/lineage_cache.h

```diff
@@ -20,16 +20,14 @@ namespace raylet {
 enum class GcsStatus {
   /// The task is not in the lineage cache.
   NONE = 0,
-  /// The task is being executed or created on a remote node.
-  UNCOMMITTED_REMOTE,
-  /// The task is waiting to be executed or created locally.
-  UNCOMMITTED_WAITING,
-  /// The task has started execution, but the entry has not been written to the
-  /// GCS yet.
-  UNCOMMITTED_READY,
-  /// The task has been written to the GCS and we are waiting for an
-  /// acknowledgement of the commit.
+  /// The task is uncommitted. Unless there is a failure, we will expect a
+  /// different node to commit this task.
+  UNCOMMITTED,
+  /// We flushed this task and are waiting for the commit acknowledgement.
   COMMITTING,
+  // TODO(swang): Add a COMMITTED state for the following scenario: We
+  // received a commit acknowledgement for this task, but a task in its
+  // lineage has not yet been committed.
```

Comment:
Why don't we need this state in this PR?

Reply:
It's just a performance optimization to avoid subscribing to tasks that we already know have been committed. I'll update the TODO to reflect/clarify this.